src/Processors/Formats/Impl/ORCBlockInputFormat.cpp

```diff
@@ -38,37 +37,28 @@ Chunk ORCBlockInputFormat::generate()
     if (!file_reader)
         prepareReader();
 
+    std::shared_ptr<arrow::RecordBatchReader> batch_reader;
+    arrow::Status reader_status = file_reader->NextStripeReader(DBMS_DEFAULT_BUFFER_SIZE, include_indices, &batch_reader);
+    if (!reader_status.ok())
+        throw ParsingException(ErrorCodes::CANNOT_READ_ALL_DATA, "Failed to create batch reader: {}", reader_status.ToString());
     if (!batch_reader)
-    {
-        arrow::Status reader_status = file_reader->NextStripeReader(
-            DBMS_DEFAULT_BUFFER_SIZE, include_indices, &batch_reader);
-        if (!reader_status.ok())
-            throw ParsingException(ErrorCodes::CANNOT_READ_ALL_DATA,
-                                    "Failed to create batch reader: {}",
-                                    reader_status.ToString());
-        if (!batch_reader)
-            return res;
-    }
+        return res;
 
-    std::shared_ptr<arrow::RecordBatch> batch_result;
-    arrow::Status batch_status = batch_reader->ReadNext(&batch_result);
-    if (!batch_status.ok())
-        throw ParsingException(ErrorCodes::CANNOT_READ_ALL_DATA,
-                                "Error while reading batch of ORC data: {}",
-                                batch_status.ToString());
+    std::shared_ptr<arrow::Table> table;
+    arrow::Status table_status = batch_reader->ReadAll(&table);
+    if (!table_status.ok())
+        throw ParsingException(ErrorCodes::CANNOT_READ_ALL_DATA, "Error while reading batch of ORC data: {}", table_status.ToString());
 
-    if (!batch_result || !batch_result->num_rows())
+    if (!table || !table->num_rows())
         return res;
 
     ArrowColumnToCHColumn::NameToColumnPtr name_to_column_ptr;
     for (const auto & column_name : column_names)
     {
-        arrow::ArrayVector vec = {batch_result->GetColumnByName(column_name)};
-        std::shared_ptr<arrow::ChunkedArray> arrow_column = std::make_shared<arrow::ChunkedArray>(vec);
+        std::shared_ptr<arrow::ChunkedArray> arrow_column = table->GetColumnByName(column_name);
         name_to_column_ptr[column_name] = arrow_column;
     }
     arrow_column_to_ch_column->arrowColumnsToCHChunk(res, name_to_column_ptr);
```

Comment:
Now we can use just `arrow_column_to_ch_column->arrowTableToCHChunk(res, table);`